src/main/java/nextstep/subway/favorite/application/FavoriteService.java

```diff
@@ -0,0 +1,47 @@
+package nextstep.subway.favorite.application;
+
+import nextstep.subway.favorite.domain.Favorite;
+import nextstep.subway.favorite.domain.FavoriteRepository;
+import nextstep.subway.favorite.dto.FavoriteRequest;
+import nextstep.subway.favorite.dto.FavoriteResponse;
+import nextstep.subway.member.application.MemberService;
+import nextstep.subway.member.domain.Member;
+import nextstep.subway.station.application.StationService;
+import nextstep.subway.station.domain.Station;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class FavoriteService {
+    private final FavoriteRepository favoriteRepository;
+    private final MemberService memberService;
+    private final StationService stationService;
+
+    public FavoriteService(
+            FavoriteRepository favoriteRepository, MemberService memberService, StationService stationService) {
+        this.favoriteRepository = favoriteRepository;
+        this.memberService = memberService;
+        this.stationService = stationService;
+    }
+
+    public FavoriteResponse create(long loginMemberId, FavoriteRequest request) {
+        Member member = memberService.findById(loginMemberId);
+        Station source = stationService.findStationById(request.getSourceStationId());
+        Station target = stationService.findStationById(request.getTargetStationId());
+        Favorite persistFavorite = favoriteRepository.save(new Favorite(member, source, target));
+        return FavoriteResponse.of(persistFavorite);
+    }
+
+    public List<FavoriteResponse> list(long loginMemberId) {
```

Comment:
조회만 있을 경우에는 readOnly 속성을 이용하면 JPA 내부적으로 조금 더 최적화되어서 다르게 동작하는 부분이 있으니
별도로 찾아보시면 도움 되실 것 같습니다

```suggestion

    @Transactional(readOnly = true)
    public List<FavoriteResponse> list(long loginMemberId) {
```

Reply:
[6ebfeca](https://github.com/next-step/atdd-subway-service/pull/628/commits/6ebfeca2bc504e102fd159d5c32862010e1a5369) 네. 이 커밋에서 조회만 이루어지는 경우에는 읽기 전용 설정을 해주었습니다.